velox/common/caching/SsdFile.cpp

```diff
@@ -866,9 +920,17 @@ T readNumber(std::ifstream& stream) {
 } // namespace
 
 void SsdFile::readCheckpoint(std::ifstream& state) {
-  char magic[4];
-  state.read(magic, sizeof(magic));
-  VELOX_CHECK_EQ(::strncmp(magic, kCheckpointMagic, 4), 0);
+  char versionMagic[4];
+  state.read(versionMagic, sizeof(versionMagic));
+  const auto checkpoinHasChecksum =
+      isChecksumEnabledOnCheckpoint(std::string(versionMagic, 4));
+  if (checksumReadVerificationEnabled_ && !checkpoinHasChecksum) {
```

Comment:
I think we shall check checksumEnabled_ here?

Reply:
Why is that?

I think in the following case, we still want to recover cache from checkpoint:
-  `checkpoinHasChecksum==false`: checkpoint file doesn't have checksum.
- `checksumReadVerificationEnabled_==false`: read verification is disabled. Don't care if checksum matches.
- `checksumEnabled_==true`: new entry we'll calculate the checksum and have it write to checkpoint file.

But if we check `checksumEnabled_`, we will skip the recovery.